src/pybind/mgr/dashboard/frontend/package.json

```diff
@@ -2,12 +2,13 @@
   "name": "ceph-dashboard",
   "version": "0.0.0",
   "license": "MIT",
+  "locale": "src/locale/messages.xlf",
   "scripts": {
     "ng": "ng",
     "start": "npm run env_build && ng serve --host 0.0.0.0",
     "build": "npm run env_build && ng build",
     "env_build": "cp src/environments/environment.tpl.ts src/environments/environment.prod.ts && cp src/environments/environment.tpl.ts src/environments/environment.ts  && node ./environment.build.js",
-    "i18n": "ng xi18n --i18n-format xlf --i18n-locale en-US --output-path locale --progress=false && ngx-extractor -i 'src/**/*.ts' -f xlf -o src/locale/messages.xlf -l en-US",
+    "i18n": "ng xi18n --i18n-format xlf --i18n-locale en-US --output-path locale --progress=false && ngx-extractor -i 'src/**/*.ts' -f xlf -o $npm_package_locale -l en-US && perl -0777 -i -pe 's|(<source>)[\\s]+(.*?)[\\s]+(</source>)|$1$2$3|g' $npm_package_locale",
```

Comment:
After reading the new description I remembered why this is not a real solution to the problem.
If you write the following html:
```
<ng-container i18n>ASD</ng-container>
<ng-container i18n> ASD</ng-container>
<ng-container i18n>
  ASD
</ng-container>
```
You will still get 3 `trans-units`, each with a different id, but with the same string.

The solution to this problem would be to either run a script to fix the source code or fix the i18n extractor itself.

Reply:
Sh*t! You're totally right! And the 'misbehaviour' is not only on the polyfill part, the standard Angular i18n creates new translation units for every whitespace change (even in interpolated parts, which is nonsensical). So... I'm afraid the only possible solution is: either using linters to enforce adherence to coding guidelines, or (my preferred) move the HTML formatting script to the html code. I'll look at this.

As a workaround in the meantime, by specifying named ids `< ... i18n="..@@named_id" ...>` we might temporarily avoid this issue.

Reply:
So, as agreed, I'll remove the i18n perl regex script and fix the newlines within i18n elements.

To follow the current status of Angular i18n implementation, let's keep an eye on https://github.com/angular/angular/issues/16477

Reply:
Btw, [these are the lines](https://github.com/angular/angular/blob/f8c7c3c09c38089558aa95a1f9c5a297fed852a6/packages/compiler/src/i18n/digest.ts#L13-L15)  that generate the trans-unit ID:
```ts
export function digest(message: i18n.Message): string {
  return message.id || sha1(serializeNodes(message.nodes).join('') + `[${message.meaning}]`);
}
```